identity-wallet/src/state/profile_settings/reducers/update_sorting_preference.rs

```diff
@@ -58,45 +63,401 @@ pub async fn sort_credentials(state: AppState, _action: Action) -> Result<AppSta
     let mut credentials: Vec<DisplayCredential> = state.credentials.clone();
     let preferences: Preferences<CredentialSortMethod> = state.profile_settings.sorting_preferences.credentials.clone();
 
-    let name_az = |a: &DisplayCredential, b: &DisplayCredential| a.display_name.cmp(&b.display_name);
-    let issuance_new_old =
-        |a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_issued.cmp(&b.metadata.date_issued);
-    let added_new_old =
-        |a: &DisplayCredential, b: &DisplayCredential| a.metadata.date_added.cmp(&b.metadata.date_added);
+    let display_names: Vec<String> = credentials
+        .iter()
+        .map(|credential| credential.display_name.clone())
+        .collect();
+    let sorted_display_names = sort(display_names, state.profile_settings.locale.clone());
 
-    credentials.sort_by(match preferences.sort_method {
-        CredentialSortMethod::NameAZ => name_az,
-        CredentialSortMethod::IssueDateNewOld => issuance_new_old,
-        CredentialSortMethod::AddedDateNewOld => added_new_old,
-    });
+    let name_az = |a: &DisplayCredential, b: &DisplayCredential| {
+        let cmp_a = sorted_display_names
+            .iter()
+            .position(|display_name| display_name == &a.display_name)
+            .unwrap();
```

Comment:
The `unwrap` here and in line 80 can be removed. `Option<usize>` can be compared as well and removing those unwraps removes any chance there might be of a panic.